docs/07.CODING-GUIDELINES.md

```diff
@@ -0,0 +1,610 @@
+# Coding style guidelines for the JerryScript project
+
+This text is a brief overview of JerryScript coding style.
+Each rule starts with a short description followed
+by several examples. We believe examples are better than
+long explanations. Please follow these guidelines when
+you submit a patch for review.
+
+## General rules
+
+* Indentation is two spaces.
+* Tab characters are not allowed.
+* Maximum line length is 120 characters long (excluding newline).
+* No trailing white spaces are allowed.
+* Run ```tools/run-tests.py --check-vera``` to check majority of the style rules.
+
+## Comments
+
+Only /* */ comments are allowed in JerryScript. Comments should
+be complete sentences, except for return value, field and
+argument descriptions (see these excpetions later). The text
+part of a comment should focus on explaing "why" rather than
+"what".
+
+```diff
++good
+```
+
+```c
+  /* A perfect comment. */
+
+  /* A perfect multiline
+   * comment. Each line should
+   * start with an asterisk. */
+```
+
+```diff
+-bad
+```
+
+```c
+  // Double slash comments are not allowed.
+
+  /* not a complete sentence */
+
+  /* A bad mulitiline
+     comment. */
+```
+
+All types, constants, and functions require a description in
+JerryScript. These comments should start with /**. The starting
```

Comment:
The `/**.` and `/**` will make the string between them bold, so what you wanted to write disappears.
example: /**. my text /**  in raw: `/**. my text /**`.
IMHO we should but such special characters in the description between backticks (`).